ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -40,97 +40,171 @@ const styles = reactStyles({
-                decrementButtonClassName='p-button-secondary'
-                incrementButtonClassName='p-button-secondary'
-                value={updatedDiskSize || masterDiskSize}
-                inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
-                onChange={({value}) => onChange(value === masterDiskSize ? null : value)}
-                min={50 /* Runtime API has a minimum 50GB requirement. */}/>
-  </div>;
+const DiskSizeSelection = ({onChange, updatedDiskSize, diskSize}) => {
+  return <Fragment>
+    <label htmlFor='runtime-disk' style={{marginRight: '.25rem'}}>Disk (GB)</label>
+    <InputNumber id='runtime-disk'
+      showButtons
+      decrementButtonClassName='p-button-secondary'
+      incrementButtonClassName='p-button-secondary'
+      value={updatedDiskSize || diskSize}
+      inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+      onChange={({value}) => onChange(value)}
+      min={50 /* Runtime API has a minimum 50GB requirement. */}/>
+  </Fragment>;
+};
+
+const DataProcConfig = ({onChange, dataprocConfig}) => {
+  const {
+    workerMachineType = defaultMachineType,
+    workerDiskSize = 50,
+    numberOfWorkers = 2,
+    numberOfPreemptibleWorkers = 0
+  } = dataprocConfig || {};
+  const [updatedNumWorkers, setUpdatedNumWorkers] = useState(numberOfWorkers);
+  const [updatedPreemtible, setUpdatedPreemptible] = useState(numberOfPreemptibleWorkers);
+  const [updatedWorkerMachine, setUpdatedWorkerMachine] = useState(null);
+  const [updatedDiskSize, setUpdatedDiskSize] = useState(workerDiskSize);
+
+  useEffect(() => {
+    const machineType = updatedWorkerMachine && updatedWorkerMachine.name;
+    const dataprocConfigChanged = updatedNumWorkers !== numberOfWorkers ||
+    updatedPreemtible !== numberOfPreemptibleWorkers ||
+    updatedDiskSize !== workerDiskSize ||
+    updatedWorkerMachine ||
+    !dataprocConfig;
+
+    onChange(dataprocConfigChanged ? {
+      workerMachineType: machineType,
+      workerDiskSize: updatedDiskSize,
+      numberOfWorkers: updatedNumWorkers,
+      numberOfPreemptibleWorkers: updatedPreemtible
+    } : null);
+
+    return () => onChange(null);
+  }, [updatedNumWorkers, updatedPreemtible, updatedWorkerMachine, updatedDiskSize]);
+
+
+  return <fieldset style={{marginTop: '0.75rem'}}>
+    <legend style={styles.workerConfigLabel}>Worker Config</legend>
+    <div style={styles.formGrid}>
+      <label htmlFor='num-workers' style={{marginRight: '.25rem'}}>Workers</label>
+      <InputNumber id='num-workers'
+        showButtons
+        decrementButtonClassName='p-button-secondary'
+        incrementButtonClassName='p-button-secondary'
+        value={updatedNumWorkers}
+        inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+        onChange={({value}) => {
+          setUpdatedNumWorkers(value);
+          if (updatedNumWorkers < updatedPreemtible) {
+            setUpdatedPreemptible(updatedNumWorkers);
+          }
+        }}
+        min={2}/>
```

Comment:
If it works - I'd like to use 0 as the min here. Note that we decided to deviate from Terra UI in the mocks by not having an option of "Spark master node" - this is because it's redundant with just setting workers to 0. The "2" requirement is throwing me off - I don't think there's any reason we can't have a 1 worker cluster (though it's not very useful), so I'm not sure why that was the chosen minimum.

Reply:
Update: lets just go with what you have now (min=2). Apparently the rules are : `It’s also not possible to have a cluster with only preemptible nodes in it. You must have at least 3 regular nodes: the master and the first 2 workers.`. I don't want to confuse users by offering this edge case third configuration, when we have GCE already available to them as an option.

If it gets requested later, probably we can just go with the Terra UI route, and add the "master only" version to the dropdown.